odoo/addons/base/models/ir_attachment.py

```diff
@@ -336,7 +336,10 @@ def _postprocess_contents(self, values):
                     nw, nh = map(int, max_resolution.split('x'))
                     if w > nw or h > nh:
                         img = img.resize(nw, nh)
-                        quality = int(ICP('base.image_autoresize_quality', 80))
+                        if _subtype == 'png':
+                            quality = 0  # Do not affect PNGs color palette
+                        else:
+                            quality = int(ICP('base.image_autoresize_quality', 80))
```

Comment:
This will make the `quality` param only used for `jpeg`, so we might want to make that explicit to avoid the guessing game, rather than discarding `png`? Especially since the value of 80 looks tailored to `jpeg` specifically.

Reply:
1. If it does not impact the others (bmp, etc), should not the code say `!== jpeg` @agau-odoo ?
2. Not sure it's really that needed to make it more explicit... are not those ICP you can only know about looking at this code? But I guess we can rename it in master if we merge this... but I feel like we only risk annoying the 3 people that modified this in their DB, for nothing, so I would not do it :shrug: I am not a format expert but I don't think any dev would be unhappy that "quality" only applies for jpeg given our supported formats (not sure about webp though)?